File: PlaybackLab/MultipleLayoutsTVSampleApp/MultipleLayoutsTVSampleApp/Lists/VideoTableViewController.m
//
//  VideoTableViewController.m
//  MultipleLayoutsTVSampleApp
//
//  Copyright © 2016 Ooyala. All rights reserved.
//

#import "VideoTableViewController.h"
#import "AbstractPlayerViewController.h"
#import "FullscreenPlayerViewController.h"
#import "PlayerSelectionOption.h"

@interface VideoTableViewController ()

@end

@implementation VideoTableViewController

- (void)viewDidLoad
{
  [super viewDidLoad];
}

#pragma mark - Table view data source

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.assets.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
  UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"OptionCell" forIndexPath:indexPath];
  
  PlayerSelectionOption *asset = self.assets[indexPath.row];
  cell.textLabel.text = asset.title;
  
  return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
  NSString *segueId = @"";
  PlayerSelectionOption *asset = self.assets[indexPath.row];
  if ([asset.hostVC isSubclassOfClass:[FullscreenPlayerViewController class]]) {
    segueId = @"fullscreenSegue";
  } else if ([asset.hostVC isSubclassOfClass:[AbstractPlayerViewController class]]) {
    segueId = @"childSegue";
  }
  
  [self performSegueWithIdentifier:segueId sender:[self.tableView cellForRowAtIndexPath:indexPath]];
}


#pragma mark - Navigation

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
  NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
  UIViewController *destinationVC = segue.destinationViewController;
  if ([destinationVC isKindOfClass:[FullscreenPlayerViewController class]]) {
    ((FullscreenPlayerViewController *)destinationVC).option = self.assets[indexPath.row];
  } else if ([destinationVC isKindOfClass:[AbstractPlayerViewController class]]) {
    ((AbstractPlayerViewController *)destinationVC).option = self.assets[indexPath.row];
  }
}

@end

File: PlaybackLab/TVOSSwiftSampleApp/TVOSSwiftSampleApp/List/VideoTableViewController.swift
//
//  ViewController.swift
//  TVOSSwiftSampleApp
//
//  Copyright © 2018 Ooyala. All rights reserved.
//

import UIKit

class VideoTableViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
  
  private var tableOptions = OptionDataSource.options
  private lazy var tableView: UITableView = {
    let tableView = UITableView(frame: self.view.bounds)
    tableView.dataSource = self
    tableView.delegate = self
    return tableView
  }()
  private var rowHeight: CGFloat = 100

  override func viewDidLoad() {
    super.viewDidLoad()

    self.view.addSubview(tableView)
  }

  override func viewWillLayoutSubviews() {
    super.viewWillLayoutSubviews()
    tableView.frame = view.bounds
  }

  // MARK: - UITableViewDelegate
  func tableView(_ tableView: UITableView,
                 cellForRowAt indexPath: IndexPath) -> UITableViewCell {
    let cell = UITableViewCell(style: .default, reuseIdentifier: nil)
    let option = tableOptions[indexPath.row]
    cell.textLabel?.text = option.title
    
    return cell
  }
  
  func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
    return rowHeight
  }
  
// MARK: - UITableViewDataSource's methods
  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
    return tableOptions.count
  }
  
  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath){
    let option = tableOptions[indexPath.row]
    performSegue(withIdentifier: option.segueName, sender: tableView.cellForRow(at: indexPath))
  }
  
  override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
    guard let indexPath = tableView.indexPathForSelectedRow else { return }
    switch segue.identifier {
    case "fullscreenSegue":
      guard let destination = segue.destination as? FullscreenPlayerViewController else { return }
      destination.option = tableOptions[indexPath.row]
    case "childSegue":
      guard let destination = segue.destination as? AbstractPlayerViewController else { return }
      destination.option = tableOptions[indexPath.row]
    case "playerTokenSegue":
      guard let destination = segue.destination as? OoyalaPlayerTokenPlayerViewController else { return }
      destination.option = tableOptions[indexPath.row]
    default:
      guard let destination = segue.destination as? FairplayPlayerViewController else { return }
      destination.option = tableOptions[indexPath.row]
    }
  }
}

